### PR TITLE
Hmmer & Stockholm Datatypes

### PR DESCRIPTION
**Pull request by hexylena**
Hmmer and stockholm datatypes that will be useful with the HMMER tool suite. https://github.com/galaxyproject/tools-iuc/tree/hmmer/tools/hmmer3

CC @bgruening
xref https://github.com/galaxyproject/tools-iuc/issues/125

### CODE REVIEW

File: lib/galaxy/datatypes/msa.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+from galaxy.datatypes.metadata import MetadataElement
+import subprocess
+import os
+
+
+import logging
+log = logging.getLogger(__name__)
+
+
+def count_special_lines( word, filename, invert=False ):
```
Comment (bgruening) [inline lib/galaxy/datatypes/msa.py]: Can we move this function to some `util` file? I used this a lot also in other [datatypes](https://github.com/bgruening/galaxytools/blob/master/chemicaltoolbox/datatypes/molecules.py) as well.

**Comment by bgruening** [comment]
I have no objections here. You could also include the other MSA datatypes if you like :)

File: config/datatypes_conf.xml.sample [hunk 2]
```diff
@@ -333,6 +336,8 @@
     <sniffer type="galaxy.datatypes.images:Xpm"/>
     <sniffer type="galaxy.datatypes.images:Eps"/>
     <sniffer type="galaxy.datatypes.images:Rast"/>
+    <sniffer type="galaxy.datatypes.msa:Stockholm_1_0" />
```
Comment (bgruening) [inline config/datatypes_conf.xml.sample]: Someone with more experience should comment here, but I guess nothing should come after the image datatypes.

**Comment by hexylena** [comment]
@bgruening I'd rather not add them in since they don't have sniffers just yet. I'll add sniffers when I can and open another PR :)

File: lib/galaxy/datatypes/msa.py (hunk 1 above)
Comment (hexylena) [inline lib/galaxy/datatypes/msa.py]: Definitely. I'll update the PR shortly.

File: config/datatypes_conf.xml.sample (hunk 2 above)
Comment (hexylena) [inline config/datatypes_conf.xml.sample]: Moved them upwards. https://github.com/erasche/galaxy/commit/d1c1af3bff2fa8959c09cdd26523dcf9c0e839a7

**Comment by peterjc** [comment]
I'm not 100% sure HMMER is a multiple-sequence alignment format (being only pairwise alignments), but that is nitpicking.

For reference: BioPerl/Biopython consider the pairwise alignments from HMMER, BLAST, etc (two aligned sequences with scoring) slightly differently to multiple sequence alignments (generally more than two sequences). See `AlignIO` vs `SearchIO`.

Otherwise +1 on adding these formats.

File: config/datatypes_conf.xml.sample (hunk 2 above)
Comment (nsoranzo) [inline config/datatypes_conf.xml.sample]: Looks like Stockholm_1_0 is still after the images sniffers.

**Comment by hexylena** [comment]
@peterjc you are correct, HMMs are not MSAs at all. SearchIO is definitely the correct decision as far as being technically correct goes. I'm open to suggestions for a new file name. :)

File: config/datatypes_conf.xml.sample (hunk 2 above)
Comment (hexylena) [inline config/datatypes_conf.xml.sample]: Thanks! Can't believe I missed that one.

**Comment by jmchilton** [comment]
Opened a PR against erasche/hmmer with a small bug fix. After that is merged I'm :+1: on this PR.

**Comment by hexylena** [comment]
Thanks! I'll release the HMMER tools now :)

**Pull request merged**